pkg/util/net/client.go

```diff
@@ -22,14 +23,22 @@ func (ht headerTransport) RoundTrip(r *http.Request) (*http.Response, error) {
 // NewHeaderClient returns an http.Client. It has the same behavior as
 // DefaultClient, except that the key-value pairs in extras are added as headers
 // on every request.
-func NewHeaderClient(extras map[string]string) *http.Client {
+func NewHeaderClient(extras map[string]string, tlsConfig *tls.Config) *http.Client {
```

Comment:
Exposing only the TLS config seems rather arbitrary. Instead of accepting at TLS config and building the inner transport from that, what about accepting an `http.RoundTripper` and letting the caller decide how to initialize it?  Or perhaps a `*http.RoundTripper`, so that `nil` can be provided to get a sensible default?